Engines/Wine/QuickScript/GoG Script/script.js

```diff
@@ -0,0 +1,70 @@
+include(["engines", "wine", "quick_script", "quick_script"]);
+include(["utils", "functions", "net", "download"]);
+include(["engines", "wine", "engine", "object"]);
+
+function GogScript() {
+    QuickScript.call(this);
+}
+
+GogScript.prototype = Object.create(QuickScript.prototype);
+
+GogScript.prototype.constructor = GogScript;
+
+GogScript.prototype.gogSetupFileName = function(setupFileName) {
+    this._setupFileName = setupFileName;
+    return this;
+}
+
+/**
+ * Presents a Gog.com login window to the user, login to its account and return a token that can be used later.
+ * Stores the tocken in a parameter
+ * @param {SetupWizard} setupWizard The setupWizard to use
+ * @returns {GogScript} This
+ */
+GogScript.prototype.loginToGog = function (setupWizard) {
+    var browserWindow = setupWizard.createBrowser("Please login to your GoG.com account so that we can download the game for you:");
+    browserWindow.goToUrl("https://auth.gog.com/auth?client_id=46899977096215655&redirect_uri=https%3A%2F%2Fembed.gog.com%2Fon_login_success%3Forigin%3Dclient&response_type=code&layout=client2");
+    browserWindow.waitForUrl("https://embed.gog.com/*");
+    var currentUrl = browserWindow.getCurrentUrl();
+    var code = currentUrl.split("code=")[1].split("&")[0];
+
+    var tokenUrl = "https://auth.gog.com/token?client_id=46899977096215655&client_secret=9d85c43b1482497dbbce61f6e4aa173a433796eeae2ca8c5f6129f2dc4de46d9&grant_type=authorization_code&code="+code+"&redirect_uri=https%3A%2F%2Fembed.gog.com%2Fon_login_success%3Forigin%3Dclient";
+    this._token = new Downloader().url(tokenUrl).json();
+    return this;
+}
+
+GogScript.prototype.download = function(setupWizard) {
+    this._setupFile = createTempFile("exe");
```

Comment:
Are you sure about this temp file name?
`exe` doesn't sound like an informative file name^^

Reply:
The first parameter of  createTmpFile is actually the extension of the file, not its name. 
The name is chosen by the framework

Reply:
Ah I forgot that.